### PR TITLE
Enable SmartThings subscribe/unsubscribe with callback urls + device graph support

### PR DESCRIPTION
**Pull request by m-chau**

### CODE REVIEW

File: org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js [hunk 1]
```diff
@@ -93,7 +94,7 @@ class Translator {
         var putPayloadString = JSON.stringify(putPayload);
         return this._makeRequest(endpointUri, this._updatePath + '/' + deviceId, 'PUT', putPayloadString)
             .then((result) => {
-                if (result === "succeed") {
```
Comment (arjun-msft) [inline org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js]: why is this chance necessary?

File: org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js [hunk 2]
```diff
@@ -115,16 +116,18 @@ class Translator {
      * and on the hub translator (this) for verification.
      */
     _subscribe(subscriptionInfo) {
-        var requestPath = '/subscription/' + subscriptionInfo.controlId;
-        return this._makeRequest(subscriptionInfo.endpointUri, requestPath, 'POST', '');
+        var requestPath = this._subcriptionPath + '?deviceId=' + subscriptionInfo.controlId 
+                                            + '&subscriptionURL=' + subscriptionInfo.callbackUrl;
+        return this._makeRequest(subscriptionInfo.endpointUri, requestPath, 'POST', "");
```
Comment (arjun-msft) [inline org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js]: did you intend to use the passed in uri as the uri to call? why would the end user have this info?

**Comment by arjun-msft** [comment]
:shipit:

File: org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js (hunk 2 above)
Comment (m-chau) [inline org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js]: I was told that whatever call the subcribe would include callbackUrl in the subscriptionInfo.

---
In reply to: [109980337](https://github.com/openT2T/translators/pull/225#discussion_r109980337) [](ancestors = 109980337)

File: org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js (hunk 1 above)
Comment (m-chau) [inline org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js]: This is align with the status return from POST methods returned from SmartApp.

---
In reply to: [109978582](https://github.com/openT2T/translators/pull/225#discussion_r109978582) [](ancestors = 109978582)

**Comment by m-chau** [comment]
@KURTIS is added to the review. #Closed

**Comment by m-chau** [comment]
I accidently added the device graph change into this PR. @danielrischar Is it possible for you to do a quick code review on this? Thank you. -M

**Pull request merged**